### DISCUSSION TITLE
How to make withAuthUser not redirect anonymous user when using "REDIRECT_TO_APP"

### DISCUSSION THREAD

**Discussion by chrsep** [gladly-team/next-firebase-auth#584] [Q&A]
Hi, I have an app that requires every client to at least be anonymously signed in. When I use REDIRECT_TO_APP on the login page like the code below, seems like next-firebase-auth always considers anonymous users to be signed in.

```js
export default withAuthUser({
  whenAuthed: AuthAction.REDIRECT_TO_APP,
})(LoginPage)
```

Is there a way to treat anonymous users as logged out? Thanks

**Comment by kmjennison** [comment]
My recommendation is to create your own higher-order component that's wrapped by `withAuthUser` and then extends the behavior: if the user returned by `useAuthUser` is anonymous, redirect; otherwise, render the child component. Then, use your HOC throughout the app in place of `withAuthUser`.

The `withAuthUser` HOC can provide some guidance for this:
https://github.com/gladly-team/next-firebase-auth/blob/v1.x/src/withAuthUser.js

Please let me know how this goes. This is a kind of customization for which we could definitely use better documentation.
